src/GitHub.App/Controllers/NavigationController.cs

```diff
@@ -66,7 +66,15 @@ public void LoadView(IConnection connection, ViewWithData data, Action<IView> on
             switch (data.MainFlow)
             {
                 case UIControllerFlow.PullRequestCreation:
+                    //The selected flow will change after the view is created, so save it.
+                    var previousFlow = Current?.SelectedFlow;
+
                     CreateView(connection, data, onViewLoad);
+                    if (data.Data == null && previousFlow == UIControllerFlow.PullRequestCreation)
+                    {
+                        Pop(history[history.Count-2]);
```

Comment:
During our weekly sync up, we decided to keep `Debug.Assert` here for now, but this probably needs an `if` statement here to not `Pop()` if the history is less than 2.